### PR TITLE
Configuring renovate bot

### PR DESCRIPTION
**Pull request by suztomo**
- Ignore example directory. The usage of "**" is in the default list. https://github.com/renovatebot/renovate/blob/master/renovate-schema.json#L300
- "sematic commit" to follow google-cloud-java https://github.com/googleapis/java-cloud-bom/blob/master/renovate.json#L63 . This is for the commit message.
- separateMajorMinor not to distinguish libraries-bom 2.X and 3.X.
- Guava's version scheme needs to distinguish "-jre" and "-android".
  Using https://github.com/renovatebot/config-help/issues/232#issuecomment-491826549 and https://github.com/googleapis/java-cloud-bom/blob/master/renovate.json#L15
- `ignoreUnstable: false` for google-cloud artifacts because its version carries "-alpha"

### CODE REVIEW

**Comment by suztomo** [comment]
@elharo PTAL.

**Comment by suztomo** [comment]
Okay. Closed the PRs by Renovate bot for now. We need renovate bot to confirm the configuration is correct.

@elharo  Which files and dependencies you think Renovate bot should not touch in addition to this PR?

File: renovate.json [hunk 1]
```diff
@@ -1,5 +1,22 @@
 {
   "extends": [
     "config:base"
+  ],
+  "ignorePaths": ["example-problems/**"],
```
Comment (suztomo) [inline renovate.json]: Memo: do we need to exclude "boms/cloud-oss-bom/pom.xml"?

Comment (chingor13) [reply renovate.json]: The cloud-oss-bom is probably the most important one for renovate to update

**Comment by suztomo** [comment]
boms/cloud-oss-bom/pom.xml?
There may be a problem in a new version. Elliotte wants two humans to confirm the change in the BOM member. I'll exclude `boms/cloud-oss-bom/pom.xml`.


We don't care boms/convergence-check/pom.xml.

Ignore anything that with "-snapshot" 
```
      <groupId>com.google.cloud.tools</groupId>
      <artifactId>dependencies</artifactId>
```

.circleci/config.yml and docs/Gemfile do not have version. No need to worry.

File: renovate.json (hunk 1 above)
Comment (suztomo) [reply renovate.json]: Elliotte prefers to maintain it manually; he wants 2 humans involved before modifying the content of the BOM.

File: renovate.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      "ignoreUnstable": false
+    },
+    {
+      "packagePatterns": [
+        "^com.google.cloud:libraries-bom",
+        "^com.google.cloud.tools:dependencies",
+        "^com.google.cloud.tools:linkage",
+        "^com.google.cloud.tools.opensource"
+      ],
+      "enabled": false
```
Comment (suztomo) [inline renovate.json]: They are the ones in this repository. Not enabling them.

File: renovate.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  "extends": [
+    "config:base"
+  ],
+  "ignorePaths": [
+    "example-problems/**",
+    "boms/cloud-oss-bom/pom.xml",
+    "boms/convergence-check/pom.xml"
```
Comment (elharo) [inline renovate.json]: I think you need to exclude boms/upper-bounds-check/ too.

Comment (suztomo) [reply renovate.json]: Updated this to ignore "boms/**".

**Pull request merged**